src/DataFramesMeta.jl

```diff
@@ -387,17 +386,29 @@ function transform(d::Union{AbstractDataFrame, AbstractDict}; kwargs...)
     return result
 end
 
+
 function transform(g::GroupedDataFrame; kwargs...)
     result = DataFrame(g)
     idx2 = cumsum(Int[size(g[i],1) for i in 1:length(g)])
     idx1 = [1; 1 + idx2[1:end-1]]
+
+    function spread_scalar(x::Vector, length::Int)
+        return x 
```

Comment:
what if `x` does not have length `length` (also maybe not use `length` as name as it clashes with `length` function).

Reply:
Good point on changing the `length` name. 

`length` is the number of observations in each group. So we are telling the function how many times to replicate the result of a `vector -> scalar` function.

Reply:
But this is why I am asking - in line 395 you assume that `v` returned a vector not a scalar and you keep it unchanged. What if the length of this vector is not correct?

Reply:
You just get a 

```
ERROR: ArgumentError: New columns must have the same length as old columns
```

Perhaps we should add an error about only doing only `vector -> vector (right length)` or `vector -> scalar`.

Reply:
You will not when by accident the total length of vectors is OK, but the vectors themselves are not of correct length (e.g. you have two groups of length 3, 6 in total, and vectors are of lengths 2 and 4). I think we should strictly check that if a `Vector` is returned it has the length equal to the length of the group.

Reply:
Okay, so 
```
length(x) == obs_in_group ? x : error...
```

Reply:
Looks good to me 👍.